src/main/webapp/app/shared/user-settings/ssh-settings/details/ssh-user-settings-key-details.component.ts

```diff
@@ -0,0 +1,145 @@
+                        return true;
+                    } else {
+                        this.isLoading = false;
+                        this.isCreateMode = true;
+                        return false;
+                    }
+                }),
+                concatMap((params) => {
+                    return this.accountService.getSshPublicKey(Number(params['keyId']));
+                }),
+                tap((publicKey: UserSshPublicKey) => {
+                    this.displayedSshKey = publicKey.publicKey;
+                    this.displayedKeyLabel = publicKey.label;
+                    this.displayedKeyHash = publicKey.keyHash;
+                    this.displayCreationDate = publicKey.creationDate;
+                    this.displayedExpiryDate = publicKey.expiryDate;
+                    this.displayedLastUsedDate = publicKey.lastUsedDate;
+                    this.isLoading = false;
+                }),
+            )
+            .subscribe();
+    }
+
+    ngOnDestroy() {
+        this.subscription.unsubscribe();
+    }
+
+    saveSshKey() {
+        const newUserSshKey = {
+            label: this.displayedKeyLabel,
+            publicKey: this.displayedSshKey,
+            expiryDate: this.displayedExpiryDate,
+        } as UserSshPublicKey;
+        this.accountService.addNewSshPublicKey(newUserSshKey).subscribe({
+            next: () => {
+                this.alertService.success('artemisApp.userSettings.sshSettingsPage.saveSuccess');
+                this.goBack();
+            },
+            error: (error) => {
+                const errorKey = error.error.errorKey;
+                if ([this.invalidKeyFormat, this.keyAlreadyExists, this.keyLabelTooLong].indexOf(errorKey) > -1) {
+                    this.alertService.error(`artemisApp.userSettings.sshSettingsPage.${errorKey}`);
+                } else {
+                    this.alertService.error('artemisApp.userSettings.sshSettingsPage.saveFailure');
+                }
+            },
+        });
+    }
+
+    goBack() {
+        if (this.isCreateMode) {
+            this.router.navigate(['../'], { relativeTo: this.route });
+        } else {
+            this.router.navigate(['../../'], { relativeTo: this.route });
+        }
+    }
+
+    validateExpiryDate() {
+        this.isExpiryDateValid = !!this.displayedExpiryDate?.isValid();
+    }
+
+    private setMessageBasedOnOS(os: string): void {
+        switch (os) {
+            case 'Windows':
+                this.copyInstructions = 'cat ~/.ssh/id_ed25519.pub | clip';
+                break;
+            case 'MacOS':
+                this.copyInstructions = 'pbcopy < ~/.ssh/id_ed25519.pub';
+                break;
+            case 'Linux':
+                this.copyInstructions = 'xclip -selection clipboard < ~/.ssh/id_ed25519.pub';
+                break;
+            case 'Android':
+                this.copyInstructions = 'termux-clipboard-set < ~/.ssh/id_ed25519.pub';
+                break;
```

Comment:
Does it make sense to assume the key name and, thus, generation algorithm here? `id_ed25519.pub`?

Reply:
Generally I'd agree that the algorithm should not be assumed, but since we also use this type in the documentation I think it makes sense to reference it here too.